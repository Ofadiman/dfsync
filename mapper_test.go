package main

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestShouldMapOverSliceOfStrings(t *testing.T) {
	strings := []string{"foo", "bar", "buzz"}

	result := mapper[string, string](strings, func(s string) string {
		return s + ":new"
	})

	snaps.MatchJSON(t, result)
}

func TestShouldMapOverSliceOfIntegers(t *testing.T) {
	ints := []int{1, 2, 3}

	result := mapper[int, int](ints, func(s int) int {
		return s * 2
	})

	snaps.MatchJSON(t, result)
}
