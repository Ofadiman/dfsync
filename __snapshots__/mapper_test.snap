
[TestShouldMapOverSliceOfIntegers - 1]
[
 2,
 4,
 6
]
---

[TestShouldMapOverSliceOfStrings - 1]
[
 "foo:new",
 "bar:new",
 "buzz:new"
]
---
